src/main/java/org/springframework/integration/dsl/context/IntegrationFlowContext.java

```diff
@@ -149,8 +149,11 @@ public void register(String flowId, IntegrationFlow integrationFlow, boolean aut
 	 */
 	public synchronized void remove(String flowId) {
 		if (this.registry.containsKey(flowId)) {
+			IntegrationFlow theFlow = this.registry.remove(flowId);
```

Comment:
`this.beanFactory.initializeBean(integrationFlow, flowId)` invokes `IntegrationFlowBeanPostProcessor` which returns `StandardIntegrationFlow` in case of lambda (see my previous comment).
So, we are starting it in the `IntegrationFlowContext.register()` pretty straightforward.
The case with `IntegrationFlowAdapter` can be covered as a direct `SmartLifecycle` implementation.

So, only the case when we don't have a `Lifecycle` is explicit `IntegrationFlow` implementation. See `FlowServiceTests.MyFlow`. for example. In this case we a warn:
```
if (theFlow instanceof Lifecycle) {
	((Lifecycle) theFlow).start();
}
else {
	throw new IllegalStateException("For 'autoStartup' mode the 'IntegrationFlow' " +
		"must be an instance of 'Lifecycle'.\n" +
		"Consider to implement it for [" + integrationFlow + "]. " +
		"Or start dependant components on their own.");
}
```
Does that work for you?